daemon/daemon.go

```diff
@@ -380,8 +380,22 @@ func (daemon *Daemon) restore() error {
 					}
 				}
 			}
+
 			if err := daemon.containerStart(c); err != nil {
```

Comment:
I think instead of even trying to start a container and fail, we should check for `NetworkByName` for all the networks that the container is configured with. If the call returns `ErrNoSuchNetwork`, then we can wait for discovery event.

This will make the check more explicit and also not even try to start the container until the store is up for name lookup.